src/test/java/org/gridsuite/explore/server/ExploreTest.java

```diff
@@ -413,6 +425,14 @@ public void testGetElementsMetadata() throws Exception {
         mockMvc.perform(get("/v1/explore/elements/metadata?ids=" + FILTER_UUID + "," + PRIVATE_STUDY_UUID + "," + CONTINGENCY_LIST_UUID)
                 .header("userId", USER1)
         ).andExpectAll(status().isOk());
+
+        mockMvc.perform(get("/v1/explore/elements/metadata?ids=" + FILTER_UUID + "," + FILTER_UUID_2 + "&elementTypes=FILTER")
+                .header("userId", USER1)
+        ).andExpectAll(status().isOk(), content().string("[{\"elementUuid\":\"" + FILTER_UUID + "\",\"elementName\":\"filterContingencyList\",\"type\":\"FILTER\",\"accessRights\":{\"private\":true},\"owner\":\"user1\",\"subdirectoriesCount\":0,\"description\":null,\"specificMetadata\":{\"id\":\"" + FILTER_UUID + "\"}},{\"elementUuid\":\"" + FILTER_UUID_2 + "\",\"elementName\":\"filterContingencyList2\",\"type\":\"FILTER\",\"accessRights\":{\"private\":true},\"owner\":\"user1\",\"subdirectoriesCount\":0,\"description\":null,\"specificMetadata\":{\"id\":\"" + FILTER_UUID_2 + "\"}}]"));
+
+        mockMvc.perform(get("/v1/explore/elements/metadata?ids=" + FILTER_UUID + "," + FILTER_UUID_2 + "&equipmentTypes=GENERATOR&elementTypes=FILTER")
+                .header("userId", USER1)
+        ).andExpectAll(status().isOk(), content().string("[]"));
```

Comment:
Add also a test with equipmentTypes=LINE, to verify that the response content is not empty in that case ...

Reply:
done